compiler/src/dmd/target.d

```diff
@@ -1265,11 +1265,14 @@ extern (C++) struct Target
      */
     extern (C++) bool isCalleeDestroyingArgs(TypeFunction tf) @safe
     {
+        return tf.linkage == LINK.cpp && omfobj;
+/*
         // On windows, the callee destroys arguments always regardless of function linkage,
         // and regardless of whether the caller or callee cleans the stack.
         return os == Target.OS.Windows ||
                // C++ on non-Windows platforms has the caller destroying the arguments
                tf.linkage != LINK.cpp;
```

Comment:
I'd limit this to just `tf.linkage == LINK.d`, it seems odd to be potentially calling a destructor on a parameter passed from a C/ObjC program.
```suggestion
         return os == Target.OS.Windows ||
                // C++ on non-Windows platforms has the caller destroying the arguments
                tf.linkage == LINK.d;
```